cli-plugins/manager/hooks.go

```diff
@@ -27,29 +28,36 @@ type HookPluginData struct {
 // a main CLI command was executed. It calls the hook subcommand for all
 // present CLI plugins that declare support for hooks in their metadata and
 // parses/prints their responses.
-func RunCLICommandHooks(dockerCli command.Cli, rootCmd, subCommand *cobra.Command, cmdErrorMessage string) {
+func RunCLICommandHooks(ctx context.Context, dockerCli command.Cli, rootCmd, subCommand *cobra.Command, cmdErrorMessage string) {
 	commandName := strings.TrimPrefix(subCommand.CommandPath(), rootCmd.Name()+" ")
 	flags := getCommandFlags(subCommand)
```

Comment:
@Benehiko @laurazard Sorry, late to the party, and not 100% sure what's best (things get somewhat confusing 😅)

Which context should be used in these cases? I see we already have the `rootCmd` and `subCommand` passed; both of which carry a context; should that context be the one to use? (`rootCmd.Context()` or `subCommand.Context()`)

(If that's the one to use, then possibly the signature wouldn't have to be changed 🤔

Reply:
that's true, although adding the ctx parameter is way more explicit and the recommended way according to the Go [documentation](https://go.dev/blog/context). The context stored inside the `rootCommand` and `subCommand` is the same as the one passed down. While I was looking at this I actually wanted to refactor the signature into an interface or using only the `docker.Cli` interface since passing down rootCmd and subCommand seems superfluous. But this is a topic for another time I think